consensus/clique/clique.go

```diff
@@ -351,7 +352,7 @@ func (c *Clique) snapshot(chain consensus.ChainReader, number uint64, hash commo
 		headers []*types.Header
 		snap    *Snapshot
 	)
-	for snap == nil {
+	for {
```

Comment:
Although you are right, if nonetheless feel safer to leave this as is, it's a lot more future proof. It also makes the code more readable, as you can see that we're looping as long as `snap` is `nil`, vs. an infinite loop, for which you'd need to dig up the exit condition inside.